pkg/monitor/cluster/nodeconditions.go

```diff
@@ -57,3 +69,30 @@ func (mon *Monitor) emitNodeConditions(ctx context.Context) error {
 
 	return nil
 }
+
+// getSpotInstances returns a map where the keys are the machine name and only exist if the machine is a spot instance
+func (mon *Monitor) getSpotInstances(ctx context.Context) map[string]struct{} {
+	spotInstances := make(map[string]struct{})
+	machines, err := mon.maocli.MachineV1beta1().Machines("openshift-machine-api").List(ctx, metav1.ListOptions{})
+
+	if err != nil {
+		// when this call fails we may report spot vms as non spot until the next successful call
```

Comment:
Would it make sense to error here and exit from `emitNodeConditions`? In this case we will not submit metrics on error at all.

I'm on the fence here, but I think having incorrect data is worse than missing data. What do you and others think?

Same question appleis to the code below (errors from `json.Unmarshal`).

Reply:
I'm on the fence as well and am open to suggestions for this. I agree that incorrect data is worse than missing data, but do we have an alert based on missing data? the worst case with incorrect data here is that we might fire a nodeready alert on a spot vm where as the worst case of missing data is we will not fire any alert when we would want one for nodeready.